src/server/qgsserviceregistry.cpp

```diff
@@ -230,9 +340,38 @@ void QgsServiceRegistry::init( const QString &nativeModulePath, QgsServerInterfa
 void QgsServiceRegistry::cleanUp()
 {
   // Release all services
-  mVersions.clear();
+  mServiceVersions.clear();
   mServices.clear();
+  mApis.clear();
   mNativeLoader.unloadModules();
 }
 
+void QgsServiceRegistry::registerApi( QgsServerApi *api )
+{
+
+  QString name    = api->name();
+  QString version = api->version();
+
+  // Test if service is already registered
+  QString key = makeServiceKey( name, version );
+  if ( mApis.constFind( key ) != mApis.constEnd() )
+  {
+    QgsMessageLog::logMessage( QStringLiteral( "Error API %1 %2 is already registered" ).arg( name, version ) );
+    return;
```

Comment:
This method could return a boolean to indicate if the API has been correctly registered (or not).